runtime/local/local_test.go

```
@@ -0,0 +1,19 @@
+package local
+
+import (
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestEntrypoint(t *testing.T) {
+	wd, _ := os.Getwd()
+
+	result := entrypoint(filepath.Join(wd, "test"))
+	assert.Equal(t, result, "cmd/main.go", "Expected entrypoint to return cmd/main.go")
+
+	result = entrypoint(filepath.Join(wd, "test/foo"))
+	assert.Equal(t, result, "main.go", "Expected entrypoint to return main.go")
```

I'm not familiar with how `runtime.Service.Source` gets populated. If I do `micro run .` from root dir but main.go lives in `cmd/service/main.go` what do we expect to pass to `entrypoint`? do we pass the root folder or `cmd/service` ?

Other tests we need to try
- cmd/servicefoo/main.go <- this is the idiomatic structure rather than just living under cmd
- multiple main.go under cmd e.g. cmd/servicefoo/main.go , cmd/servicebar/main.go <- in this scenario we can't make a decision of what to run so need to return an error to user

---

For the local implementation, source is the tmp dir the source exists at. I'll amend so it looks for any file within cmd and errors if more than one is found.